### PR TITLE
angular 9

### PR DESCRIPTION
**Pull request by RajeevSiewnath**

### CODE REVIEW

File: src/app/contacts/contacts.component.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    super();
+  }
+
+  ngOnInit(): void {
+    this.subscribe(
+      this.memberService.subject.subscribe(member => {
+        this.member = member;
+        this.peopleService.fetchByEntityId(this.member.id);
+      }),
+      this.peopleService.subject.subscribe(people => (this.people = people.people)),
```
Comment (gbervik) [inline src/app/contacts/contacts.component.ts:42]: This list could get quite long, especially for the larger Members. In the old UI we had a search form here. Could we do something similar in the angular 9 version?

File: src/app/organization/organization.component.html [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+            </div>
+            <div class="col-4 p-2 text-right">
+              <app-panel-row-actions
+                (cancel)="this.switchEditMode(0)"
+                (save)="changeForm.onSubmit(undefined)"
+              ></app-panel-row-actions>
+            </div>
+          </div>
+        </form>
+      </ng-container>
```
Comment (gbervik) [inline src/app/organization/organization.component.html:117]: In the old UI we had a note here. Could we add that back in please?

image

File: src/app/contacts/contacts.component.ts (hunk 1 above)
Comment (RajeevSiewnath) [reply src/app/contacts/contacts.component.ts:42]: Yeah for sure, I can make the same async search function as I used in one of the other pages

**Comment by RajeevSiewnath** [comment]
> I noticed when testing that the notifications persist until manually dismissed. In our AngularJS versions of the applications the notifications disappear after short of time. What are your thoughts on that? On the one hand it can be annoying if the page jumps up every time a notification disappears, but on the other it can get annoying to manually dismiss them.
> 
> image

Yeah  it makes sense to have a timeout, and even like an animated slide up (to prevent heavy jumping). This would convenient to add in the library. Also the functions to actually add these messages are a bit cumbersome, so it would also be a good opportunity to simplify that at the same time.

File: src/app/organization/organization.component.html (hunk 2 above)
Comment (RajeevSiewnath) [reply src/app/organization/organization.component.html:117]: Sure thing!

File: src/app/contacts/contacts.component.ts (hunk 1 above)
Comment (RajeevSiewnath) [reply src/app/contacts/contacts.component.ts:42]: Actually, maybe even make a library component out of it

**Comment by fabian** [comment]
Will wait with my review until Adam's point have been addressed.

**Comment by RajeevSiewnath** [comment]
That’s good. Also I noticed some slight other bugs, but I’ll first finish the events app. Should be anyway now

**Review by gbervik — CHANGES_REQUESTED** [review]
Looks good Rajeev. 

The notifications still persist at the top of the page, but it's fine to take care of that later.

A few small things to fix where optional fields are mandatory:
- organization has website as mandatory but should be optional
- addresses Line 2, 3, and 4 are mandatory but they should be optional
- website description is mandatory but should be optional

**Comment by RajeevSiewnath** [comment]
@gbervik the api disagrees with you regarding the website being optional :D 

Screenshot 2020-09-10 at 18 16 16

Also for the website description, the API does not agree

Screenshot 2020-09-10 at 18 23 23

I made them now all optional in the UI, so you can change that now in the API

**Comment by gbervik** [comment]
> @gbervik the api disagrees with you regarding the website being optional :D
>
> Also for the website description, the API does not agree
> 
> I made them now all optional in the UI, so you can change that now in the API

Oops! 😆  
Ok, I'll change that in the API

**Comment by gbervik** [comment]
Ok, I've fixed the second error. 

The first one looks like it's a problem when editing the website for an existing organisation. In that case it's doing a PUT to update the existing website

image

When trying to create a new organisation for a Member it's failing with this in the console:

image

and it's showing this error which I can't dismiss:

image

**Comment by RajeevSiewnath** [comment]
Hmmm do you have the service running on your localhost? And how are the CROS policies set up? This does not necessarily look like an app error, but rather a server misconfiguration. We could telco later today to go through it?

**Comment by gbervik** [comment]
Yep, I was running it on localhost. Everything else seemed to be working ok. 

I tried it on the newly resurrected org.worldskills.show and it creates a new organization ok without a website URL.

Happy to chat when you're back online

**Comment by RajeevSiewnath** [comment]
member organization website should be fixed now. the website field is optional and the app will call the appropriate endpoints when creating, updating or deleting the website.

**Review by gbervik — APPROVED** [review]

**Review by TehWazzard — APPROVED** [review]

**Comment by TehWazzard** [comment]
The last 2 conflict files don't seem to exist on in the new branch.

**Comment by TehWazzard** [comment]
Looks like the files we added to support the .dev branch broke the pull request. fixed now,

**Pull request merged**